util/calc.py

```diff
@@ -217,8 +217,8 @@ def _initialize(self):
     def provides(self, key):
         return key in self.id_map
 
-    def async_query(self):
-        self.future = fetch_data(self.stream_key, self.query_time_range)
+    def async_query(self,limit = None):
```

Comment:
I think limiting the query here is the wrong approach. This will work fine on cases where we only have a single stream contributing to the derived products but in cases where there are more than a single stream involved and the second stream contains more particles than the primary stream you are going to end up with truncated data for the second stream.

I would recommend that we either:

* Apply the limit only to the primary stream or
* Apply the limit when generating particles / netcdf only, leaving the queries themselves unbounded